packages/rich-text/src/SyncEditorValue.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from 'react';
+
+import equal from 'fast-deep-equal';
+
+import { usePlateSelectors } from './internal/hooks';
+import { withoutNormalizing } from './internal/misc';
+import { isNode, getEndPoint } from './internal/queries';
+import { select } from './internal/transforms';
+import { PlateEditor, Node, Value } from './internal/types';
+
+/**
+ * Plate api doesn't allow to modify (easily) the editor value
+ * programmatically after the editor instance is created.
+ *
+ * This function is inspired by:
+ * https://github.com/udecode/plate/issues/1269#issuecomment-1057643622
+ */
+const setEditorContent = (editor: PlateEditor, nodes?: Node[]): void => {
+  // Replaces editor content while keeping change history
+  withoutNormalizing(editor, () => {
+    const children = [...editor.children];
+    children.forEach((node) => editor.apply({ type: 'remove_node', path: [0], node }));
+
+    if (nodes) {
+      const nodesArray = isNode(nodes) ? [nodes] : nodes;
+      nodesArray.forEach((node, i) => {
+        editor.apply({ type: 'insert_node', path: [i], node });
+      });
+    }
+
+    const point = getEndPoint(editor, []);
+    if (point) {
+      select(editor, point);
+    }
+  });
+};
+
+export type SyncEditorStateProps = {
+  incomingValue?: Value;
+};
+
+/**
+ * A void component that's responsible for keeping the editor
+ * state in sync with incoming changes (aka. external updates)
+ *
+ * This component is a hack to workaround the limitation of Plate v17+
+ * where we can no longer access the editor instance outside the Plate
+ * provider.
+ */
+export const SyncEditorValue = ({ incomingValue }: SyncEditorStateProps) => {
```

Comment:
`IncomingValueEditorSyncer` don't know if this sounds any better just throwing it out there

Reply:
Open to suggestions. I will rethink this when I get the PR green 😅